File: weave_introduction_handson/en/5_5_saved_views.py
"""
5_5: Saved Views - Save common Trace and Evaluation views

==========================================================
What are Saved Views?
==========================================================

Saved Views let you save filters, sorting, columns, and display settings for
Trace and Evaluation tables.

Instead of rebuilding the same view each time, you can save views for common
investigation or review workflows.

==========================================================
Common use cases
==========================================================

1. Failed trace investigation
   - Show only traces with errors
   - Filter to a specific op_name
   - Sort by high latency

2. Evaluation review
   - Show samples with low scorer values
   - Compare by model or dataset
   - Display only columns needed for review

3. Team sharing
   - Save common investigation views
   - Let team members inspect data under the same conditions

4. SDK workflows
   - Use SavedView from the Python SDK
   - Configure filters and columns programmatically

==========================================================
Basic workflow
==========================================================

In the UI:
   1. Open the Traces or Evals table
   2. Adjust filters, sorting, and columns
   3. Save the view
   4. Reopen the saved view later

In the SDK:
   1. Create `weave.SavedView()`
   2. Configure columns and filters
   3. Save the view
   4. Inspect it with `.to_grid()`

==========================================================
How this relates to this hands-on
==========================================================

As traces and evaluations grow, it becomes hard to inspect everything
manually.

Saved Views make repeated workflows reusable:
"show failures", "show low scorer examples", or "show high-latency traces".

==========================================================
Reference
==========================================================

Create and manage saved views:
https://docs.wandb.ai/weave/guides/tools/saved-views

==========================================================
Note
==========================================================

- This script only prints an overview
- Saved Views are available from both the UI and Python SDK
"""

print(__doc__)

File: weave_introduction_handson_typescript/jp/5_5_saved_views.ts
/**
 * 5_5: Saved Views - よく使う Trace / Eval の見方を保存する
 *
 * TypeScript 対応: 主に UI 機能
 * - TypeScript で生成した Trace / Eval の table view を保存できます。
 * - Python SDK の SavedView helper は、この TypeScript 版では扱いません。
 */

// =============================================================================
// SECTION: Saved Views Overview
// =============================================================================
console.log(`
5_5: Saved Views - よく使う Trace / Eval の見方を保存する

Saved Views は、Traces や Evaluations の table 表示について、
filter、sort、columns、表示設定を保存して再利用する機能です。

TypeScript 版での扱い:
- TypeScript script で attributes を付けておくと filter しやすくなります。
- Traces または Evals の table で条件を調整し、Save view で保存します。
- チームで調査条件を共有できます。

Python 版との差分:
- Python SDK の weave.SavedView() による programmatic 作成は、この教材では TypeScript 移植していません。

参考:
https://docs.wandb.ai/weave/guides/tools/saved-views
`);
